docs/wasmer-js/node-modules/examples/transforming-wasi-modules/transforming-wasi-modules.md

```diff
@@ -6,69 +6,201 @@ sidebar_label: Transforming WASI Modules
+In the previous Hello World example, we showed you how to run the very basic `as-echo` WASM module that received a text string as an argument and simply echoed it back via standard out.  However, some WASI modules may be compiled in a way that means they can't immediately be run from a JavaScript environment such as a browser.
 
-Thus, in order to run these modules, you would have to **transform the module** with `@wasmer/wasm-transformer`.
+For example, any module that calls the [clock\_time\_get](https://github.com/NuxiNL/cloudabi/blob/master/cloudabi.txt#L1230) WASI API, must be able to supply a 64-bit, signed integer.  However, passing a JavaScript `BigInt` to a WebAssembly `i64` is not yet supported &mdash; this detail is still at the proposal stage.  (See [here](https://github.com/WebAssembly/JS-BigInt-integration/issues/15) and [here](https://github.com/WebAssembly/proposals/issues/7) for details).
 
-In this example, we will fetch a WASI module, transform it with `lowerI64Imports` from `@wasmer/wasm-transformer`, and then run it in the browser!
+However, it is not impossible to run such a module; but before we can, we must first ***transform*** it using `@wasmer/wasm-transformer`.
 
-```javascript
+> ### Under The Hood  
+> Technically, this transformation adapts the interface so that JavaScript `BigInt`s (64-bit, signed integers) in can be transferred to and from WebAssembly in the form of arrays of 8, unsigned 8-bit integers (a JavaScript `Uint8Array`)
+
+## Prerequisites
+
+Make sure [Parcel](https://parceljs.org/) has been installed and is available from the command line
+
+```bash
+$ npm install -g parcel
+```
+
+> ### Mac users
+> Before the installation of Parcel will work on a Mac, you must first install the [Xcode Command Line Tools](https://developer.apple.com/download/more/?=for%20Xcode)
+
+## Run Example from Git Clone
+
+If you have not already done the previous `hello-world` example, the simplest way to run this exercise is to clone the entire [`docs.wasmer.io`](https://github.com/wasmerio/docs.wasmer.io) repo:
+
+1. Change into some development directory
+
+    ```bash
+    $ cd <some_development directory>
+    ```
+
+1. Clone this entire repo
+
+    ```bash
+    $ git clone https://github.com/wasmerio/docs.wasmer.io.git
+    ```
+
+1. Change into the `transforming-wasi-modules` directory
+
+    ```bash
+    $ cd docs.wasmer.io/docs/wasmer-js/node-modules/examples/transforming-wasi-modules
+    ```
+
+1. Install the required `npm` dependencies
+
+    ```bash
+    $ npm install
+    ```
+
+1. Start `parcel` 
+
+   ```bash
+   $ parcel index.html
+   ```
+
+1. Point your browser to [`http://localhost:1234`](http://localhost:1234) and you should see `Standard Output: Done!` appear both on the browser screen and in the JavaScript console
+
+# The `clock_time_get` WebAssembly Module
```

Comment:
So personally I don't think this should live in the docs here. We wanted to focus on running WebAssemlbly modules in these docs, rather than explaining them, and teaching WebAssembly. Teaching WebAssembly, and how to build/write it itself we were planning on doing somewhat later in its own section.

Also, since we are examining the `.wat` here, which is not really meant for an end user to have to read/write, I feel like this can be placed in a separate tutorial or something later down the line 😄 👍 

cc @syrusakbary for the judgement call here.

Reply:
@ChrisWhealy Thank you for the thoughtful  explanation 😄 You make some very good points, I'm 50 / 50 on this now. Thank you for moving it, but, if anything, we can move it back if @syrusakbary is also convinced 👍

Reply:
Sure.

I'm not too bothered ***where*** this explanation goes (put it in multiple places if you want).  I do contend the following though:
* The explanation must be provided
* The explanation should be put in a place (or places) where it would naturally arise in someone's train of thought

Reply:
Ah okay, if you are not worried where it goes, I definitely agree with the two points. I think where it currently is is good, and if people open issues, or have trouble running through the docs, lets move it / link to it 😄 👍

Reply:
👍🏼